src/live.rs

```diff
@@ -829,6 +836,30 @@ RequiredBy={install_target}",
         )
     }
 
+    pub fn installer_config(&mut self, path: &str) -> Result<()> {
+        let f = OpenOptions::new()
+            .read(true)
+            .open(path)
+            .with_context(|| format!("opening installer config {}", path))?;
+        // we don't validate but at least we parse
+        let conf = serde_yaml::from_reader::<_, InstallConfig>(f)
+            .with_context(|| format!("parsing installer config {}", path))?;
+        self.installer_config_struct(&Self::filename(path)?, &conf)
+    }
+
+    fn installer_config_struct(&mut self, filename: &str, config: &InstallConfig) -> Result<()> {
+        self.live.add_file(
+            format!(
+                "/etc/coreos/installer.d/{:04}-{}",
+                self.installer_serial, filename
+            ),
+            &serde_yaml::to_vec(config).context("serializing installer config")?,
```

Comment:
Why reserialize here? From a debugging point of view, it'd probably be better to just drop the config verbatim so e.g. comments are retained. We parsed it, so we know it's at least valid.